avaframe/in3Utils/geoTrans.py

```diff
@@ -288,35 +291,85 @@ def remeshDEM(cfg, dem):
     cszDEM = headerDEM['cellsize']
     # remesh if input DEM size does not correspond to the computational cellSize
     if np.abs(cszNew - cszDEM) > cszThreshold:
-        log.info('Remeshing the input DEM (of cell size %.4g m) to a cell size of %.4g m' % (cszDEM, cszNew))
-        x, y, xNew, yNew, diffExtentX, diffExtentY = getMeshXY(dem, cellSizeNew=cszNew)
-        xGrid, yGrid = np.meshgrid(x, y)
-        xGrid = xGrid.flatten()
-        yGrid = yGrid.flatten()
-        z = dem['rasterData']
-        zCopy = np.copy(z)
-        zCopy = zCopy.flatten()
-        mask = np.where(~np.isnan(zCopy))
-        xGrid = xGrid[mask]
-        yGrid = yGrid[mask]
-        z = zCopy[mask]
-
-        headerRemeshed = {}
-        headerRemeshed['cellsize'] = cszNew
-        headerRemeshed['ncols'] = len(xNew)
-        headerRemeshed['nrows'] = len(yNew)
-        headerRemeshed['xllcenter'] = headerDEM['xllcenter']
-        headerRemeshed['yllcenter'] = headerDEM['yllcenter']
-        headerRemeshed['noDataValue'] = headerDEM['noDataValue']
-
-        dem['header'] = headerRemeshed
-        xNewGrid, yNewGrid = np.meshgrid(xNew, yNew)
-        zNew = sp.interpolate.griddata((xGrid, yGrid), z, (xNewGrid, yNewGrid), method='cubic',
-                                       fill_value=headerDEM['noDataValue'])
-        log.info('Remeshed data extent difference x: %f and y %f' % (diffExtentX, diffExtentY))
-        dem['rasterData'] = zNew
-
-    return dem
+        # first check if remeshed DEM is available
+        remeshedDem, DEMFound = searchRemeshedDEM(cfg)
```

Comment:
maybe add an option to force the remeshing even if the re meshed dem already exists? And make it the default one.

Reply:
I added a flag forceRemesh in com1DFACfg.ini - if True force remeshing - now default